Apps/Src/BlinkLights.c

```diff
@@ -0,0 +1,93 @@
+#include "Tasks.h"
+#include "Lights.h"
+
+void Task_BlinkLight(void* p_arg){
+    //get necessary state pointers
+    car_state_t *car_state = (car_state_t*) p_arg;
+    blinker_states_t *blinkerStates = &(car_state -> BlinkerStates);
+    light_switches_t *storedLightSwitches = &(car_state -> SwitchStates.lightSwitches);
```

Comment:
Are you just using this to implement a global variable of type `car_state_t`, or am I missing something here?

Reply:
@ClarkPoon All of our threads upon creation will be given a pointer to our global car state struct which will eventually be instantiated in the main.c file. The definition for this struct can be found [here](https://github.com/lhr-solar/Controls/blob/master/Apps/Inc/CarState.h). This section is just extracting the relevant fields from the car state that pertain to this thread so we can work with them locally.

Reply:
Ok. I noticed that you aren't using a mutex for this struct. I don't see any obvious critical sections that would require a mutex in your code, so I am guessing this is intentional. 

If you are modifying the same fields of this struct in multiple places, or relying on accessing multiple fields to be atomic, I would consider adding a mutex

Reply:
I was considering the same thing (mutex locking the global state), however, as far as I know currently, each field of the car state is only modified by one thread. I am planning on inspecting the code later on to make sure that this is the case though.